### PR TITLE
Add Go SDK generation config

### PR DESCRIPTION
**Pull request by vladbarosan**

### CODE REVIEW

**Comment by vladbarosan** [comment]
Fixes https://github.com/Azure/azure-sdk-for-go/issues/1246

File: specification/security/resource-manager/readme.md [hunk 1]
```diff
@@ -65,4 +74,31 @@ csharp:
...
+```
+
+## Go
+
+These settings apply only when `--go` is specified on the command line.
+
+``` yaml $(go)
+go:
+  license-header: MICROSOFT_APACHE_NO_VERSION
+  namespace: mysql
```
Comment (jhendrixMSFT) [inline specification/security/resource-manager/readme.md]: Copy/paste error.

File: specification/security/resource-manager/readme.md [hunk 2]
```diff
@@ -65,4 +74,31 @@ csharp:
...
+  - tag: package-composite-v1
+```
+
+### Tag: package-composite-v1 and go
+
+These settings apply only when `--tag=package-composite-v1 --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag) == 'package-composite-v1' && $(go)
+output-folder: $(go-sdk-folder)/services/preview/security/mgmt/2017-08-01-preview/security
```
Comment (jhendrixMSFT) [inline specification/security/resource-manager/readme.md]: If you fix the namespace value you can replace the hard-coded ```security``` path/namespace with ```$(namespace)```.  One day I'd like to see the API version be defined along with a tag so we don't have to duplicate it here.

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by vladbarosan** [comment]
@chlahav there seems to be a problem with the swagger, are the 2 api version specs compatible ?
The following error shows up when trying to generate the Go SDK:
FATAL: System.InvalidOperationException: Found incompatible property types ,  for property 'properties' in schema inheritance chain

**Comment by lmazuel** [comment]
@vladbarosan We try to move to language configuration in its own file `readme.go.md`. Could it be done this way? Thanks!
Not critical.

**Comment by vladbarosan** [comment]
@lmazuel  sure will update either wont merge until the generation actually works

**Review by chlahav — APPROVED** [review]

**Comment by vladbarosan** [comment]
@lmazuel is there a way to re trigger the go generation to see if it works ?

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-sdk-for-go

**Comment by lmazuel** [comment]
@vladbarosan it seems it doesn't :(

**Comment by vladbarosan** [comment]
@lmazuel thanks for the command. Yes I noticed the problem is that it seems to be a modeler error which seems to point to a swagger issue `FATAL: System.InvalidOperationException: Found incompatible property types ,  for property 'properties' in schema inheritance chain`

**Comment by vladbarosan** [comment]
Merging and https://github.com/Azure/azure-rest-api-specs/pull/3453 should fix generation issues

**Pull request merged**